src/Microsoft.Health.Fhir.Core/Features/Search/Parameters/SearchParameterOperations.cs

```diff
@@ -60,64 +60,73 @@ public SearchParameterOperations(
+
+                        await _searchParameterStatusManager.AddSearchParameterStatusAsync(new List<string> { searchParameterWrapper.Url }, cancellationToken);
                     }
-                }
-
-                (bool Supported, bool IsPartiallySupported) supportedResult = _searchParameterSupportResolver.IsSearchParameterSupported(searchParameterInfo);
-
-                if (!supportedResult.Supported)
-                {
-                    throw new SearchParameterNotSupportedException(string.Format(Core.Resources.NoConverterForSearchParamType, searchParameterInfo.Type, searchParameterInfo.Expression));
-                }
-
-                // check data store specific support for SearchParameter
-                if (!_dataStoreSearchParameterValidator.ValidateSearchParameter(searchParameterInfo, out var errorMessage))
-                {
-                    throw new SearchParameterNotSupportedException(errorMessage);
-                }
-
-                _logger.LogInformation("Adding the search parameter '{Url}'", searchParameterWrapper.Url);
-                _searchParameterDefinitionManager.AddNewSearchParameters(new List<ITypedElement> { searchParam });
-
-                await _searchParameterStatusManager.AddSearchParameterStatusAsync(new List<string> { searchParameterWrapper.Url }, cancellationToken);
-            }
-            catch (FhirException fex)
-            {
-                _logger.LogError(fex, "Error adding search parameter.");
-                fex.Issues.Add(new OperationOutcomeIssue(
-                    OperationOutcomeConstants.IssueSeverity.Error,
-                    OperationOutcomeConstants.IssueType.Exception,
-                    Core.Resources.CustomSearchCreateError));
+                    catch (FhirException fex)
+                    {
+                        _logger.LogError(fex, "Error adding search parameter.");
+                        fex.Issues.Add(new OperationOutcomeIssue(
+                            OperationOutcomeConstants.IssueSeverity.Error,
+                            OperationOutcomeConstants.IssueType.Exception,
+                            Core.Resources.CustomSearchCreateError));
 
-                throw;
-            }
-            catch (Exception ex)
-            {
-                _logger.LogError(ex, "Error adding search parameter.");
-                var customSearchException = new ConfigureCustomSearchException(Core.Resources.CustomSearchCreateError);
-                customSearchException.Issues.Add(new OperationOutcomeIssue(
-                    OperationOutcomeConstants.IssueSeverity.Error,
-                    OperationOutcomeConstants.IssueType.Exception,
-                    ex.Message));
-
-                throw customSearchException;
-            }
+                        throw;
+                    }
+                    catch (Exception ex) when (!(ex is FhirException))
```

Comment:
Why is this addition needed? It is implied by this catch being after the one for FhirException.